### PR TITLE
feat(helm): add topologySpreadConstraint (#3538)

### PR DESCRIPTION
**Pull request by joke**
basic support for topologySpreadConstraint

### CODE REVIEW

**Review by aleks-p — APPROVED** [review]

**Comment by aleks-p** [comment]
Hey @joke, thanks for your contribution! It looks like you need to run `make helm/docs` before this can be merged.

Via https://github.com/grafana/pyroscope/actions/runs/10742066442/job/29798271147?pr=3539

> Documentation not up to date. Please run helm-docs and commit changes!

**Comment by joke** [comment]
> Hey @joke, thanks for your contribution! It looks like you need to run `make helm/docs` before this can be merged.
> 
> Via https://github.com/grafana/pyroscope/actions/runs/10742066442/job/29798271147?pr=3539
> 
> > Documentation not up to date. Please run helm-docs and commit changes!

done

File: operations/pyroscope/helm/pyroscope/values.yaml [hunk 1]
```diff
@@ -110,6 +110,9 @@ pyroscope:
 
   nodeSelector: {}
 
+  # -- Topology Spread Constraints
+  topologySpreadConstraints: [ ]
```
Comment (aleks-p) [inline operations/pyroscope/helm/pyroscope/values.yaml:114]: looks like we hit a minor lint error here

Comment (joke) [reply operations/pyroscope/helm/pyroscope/values.yaml]: done

**Comment by aleks-p** [comment]
It looks like we are all set, feel free to merge it at any time!

**Comment by joke** [comment]
> It looks like we are all set, feel free to merge it at any time!

I don't have permission to do that 😁

**Pull request merged**